_std/macros/actions.dm

```diff
@@ -0,0 +1,44 @@
+/**
+* creates an actionbar that calls the specified proc on !!either!! owner or target if the duration passes without interruption.
+* icon / icon state can also be set. duration should be in seconds. end message is a string that displays once the action succeeds.
+* you can directly instantiate a callback actionbar if you need to modify variables on it that the macro doesnt easily let you do.
+* for instance, callback actionbars have an "args" variable you can modify to call the callback proc with arguements.
+* just make sure to start it once its been made!
+*/
+#define SETUP_GENERIC_ACTIONBAR(owner, target, duration, owner_proc_path, target_proc_path, action_icon, action_icon_state, end_message) \
+	actions.start(new /datum/action/bar/icon/callback(owner, target, duration, owner_proc_path, target_proc_path, action_icon, \
+	action_icon_state, end_message), owner)
+
+/* example for standard use:
+
+/obj/item/foo
+	name = "foo"
+	icon = 'icons/obj/foo.dmi'
+	icon_state = "foo"
+
+	attack_self(var/mob/M)
+		M.visible_message("[M] starts fiddling with \the [src].")
+		SETUP_GENERIC_ACTIONBAR(M, src, 5 SECONDS, null, /obj/item/foo/proc/cool_proc, src.icon, src.icon_state,\
+		"[M] finishes fiddling with \the [src]")
+
+	proc/cool_proc()
+		boutput(world, "farts")
+*/
+
+/* example for alternate use:
+
+/obj/item/foo
+	name = "foo"
+	icon = 'icons/obj/foo.dmi'
+	icon_state = "foo"
+
+	attack_self(var/mob/M)
+		M.visible_message("[M] starts fiddling with \the [src].")
+		var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(M, src, 5 SECONDS, null, /obj/item/foo/proc/cool_proc,\
```

Comment:
this example doesn't seem to actually use the macro?

Reply:
clarified why i put it there!